concent_api/gatekeeper/views.py

```diff
@@ -78,36 +86,65 @@ def download(request):
+        return gatekeeper_access_denied_response(
+            "Missing 'Authorization' header.",
+            ErrorCode.HEADER_AUTHORIZATION_MISSING.value,
+            path_to_file,
+        )
 
     authorization_scheme_and_token = request.META['HTTP_AUTHORIZATION'].split(" ", 1)
     assert len(authorization_scheme_and_token) in [1, 2]
 
     if len(authorization_scheme_and_token) == 1:
-        return gatekeeper_access_denied_response("Missing token in the 'Authorization' header.", path_to_file)
+        return gatekeeper_access_denied_response(
+            "Missing token in the 'Authorization' header.",
+            ErrorCode.HEADER_AUTHORIZATION_MISSING_TOKEN.value,
+            path_to_file,
+        )
 
     (scheme, token) = authorization_scheme_and_token
 
     if scheme != 'Golem':
-        return gatekeeper_access_denied_response("Unrecognized scheme in the 'Authorization' header.", path_to_file)
+        return gatekeeper_access_denied_response(
+            "Unrecognized scheme in the 'Authorization' header.",
+            ErrorCode.HEADER_AUTHORIZATION_UNRECOGNIZED_SCHEME.value,
+            path_to_file,
+        )
 
     try:
         decoded_auth_header_content = b64decode(token, validate = True)
     except binascii.Error:
-        return gatekeeper_access_denied_response("Unable to decode token in the 'Authorization' header.", path_to_file)
+        return gatekeeper_access_denied_response(
+            "Unable to decode token in the 'Authorization' header.",
+            ErrorCode.HEADER_AUTHORIZATION_UNABLE_TO_DECODE.value,
+            path_to_file,
+        )
 
     try:
         loaded_golem_message = load(decoded_auth_header_content, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY, check_time = False)
     except MessageError:
-        return gatekeeper_access_denied_response("Token in the 'Authorization' header is not a valid Golem message.", path_to_file)
+        return gatekeeper_access_denied_response(
+            "Token in the 'Authorization' header is not a valid Golem message.",
+            ErrorCode.HEADER_AUTHORIZATION_TOKEN_NOT_VALID_MESSAGE.value,
+            path_to_file,
+        )
 
     if loaded_golem_message is None:
-        return gatekeeper_access_denied_response('Undefined golem message.', path_to_file)
+        return gatekeeper_access_denied_response(
+            'Undefined golem message.',
+            ErrorCode.MESSAGE.value,
+            path_to_file,
+        )
     assert isinstance(loaded_golem_message, Message)
 
     # Check if request header contains Concent-Client-Public-Key:
     if 'HTTP_CONCENT_CLIENT_PUBLIC_KEY' not in request.META:
-        return gatekeeper_access_denied_response('Missing Concent-Client-Public-Key header.', path_to_file, loaded_golem_message.subtask_id)
+        return gatekeeper_access_denied_response(
+            'Missing Concent-Client-Public-Key header.',
+            ErrorCode.HEADER_CLIENT_PUBLIC_KEY_MISSING.value,
+            path_to_file,
```

Comment:
This header has been removed. I think this will disappear when you rebase.

Reply:
:+1: Yes, I am currently updating this part in rebase.